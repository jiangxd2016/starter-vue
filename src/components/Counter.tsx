import type { PropType } from 'vue';
import { defineComponent, onMounted, ref } from 'vue';
import { useCounter } from '@vueuse/core';

export default defineComponent({
  props: {
    initial: {
      type: Number as PropType<number>,
      default: 0,
    },
  },
  setup(props) {
    const { count, inc, dec } = useCounter(props.initial);
    const countRef = ref<HTMLDivElement>();

    return () => (
      <div ref={countRef}>
        <button class="inc btn" onClick={() => inc()}>
          +
        </button>
        <input type="text" class="text-center" v-model={count.value} />
        <button class="dec btn" onClick={() => dec()}>
          -
        </button>
      </div>
    );
  },
});
