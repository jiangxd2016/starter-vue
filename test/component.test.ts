import { mount } from '@vue/test-utils';
import Counter from '../src/components/Counter';

describe('Counter', () => {
  it('should render', () => {
    const wrapper = mount(Counter,{ props: { initial: 10 } });
    expect(wrapper.html()).toMatchSnapshot();
    const vm = wrapper.vm
    const input = vm.$el.querySelector('input')
    expect(input.value).toContain('10');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should be interactive', async () => {
    const wrapper = mount(Counter, { props: { initial: 0 } });
    const vm = wrapper.vm
    const input = vm.$el.querySelector('input')
    expect(input.value).toContain('0');

    expect(wrapper.find('.inc').exists()).toBe(true);

    await wrapper.find(".dec").trigger('click');

    const input2 = vm.$el.querySelector('input')
    expect(input2.value).toContain('1');

  });
});
