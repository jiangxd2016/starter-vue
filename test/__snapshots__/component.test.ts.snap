// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter > should render 1`] = `"<div><button class=\\"inc btn\\">+</button><input type=\\"text\\" class=\\"text-center\\"><button class=\\"dec btn\\">-</button></div>"`;

exports[`Counter > should render 2`] = `"<div><button class=\\"inc btn\\">+</button><input type=\\"text\\" class=\\"text-center\\"><button class=\\"dec btn\\">-</button></div>"`;
